### PR TITLE
Add more context locations for Post Type related context

### PR DESCRIPTION
**Pull request by bordoni**
I noticed when doing some canonical redirect work for B20.02, I realized that some of our variables were not looking for all the correct places to determine the Post Type related variables.

Related: https://github.com/moderntribe/tribe-common/pull/1277

### CODE REVIEW

File: src/Tribe/Service_Providers/Context.php [hunk 1]
```diff
@@ -249,6 +249,15 @@ static function () {
 				],
 				'tec_post_type'        => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return count( array_filter( [
+									! empty( tribe_get_request_var( TEC::POSTTYPE, false ) ),
+									! empty( tribe_get_request_var( Venue::POSTTYPE, false ) ),
+									! empty( tribe_get_request_var( Organizer::POSTTYPE, false ) ),
+								] ) );
```
Comment (lucatume) [inline src/Tribe/Service_Providers/Context.php]: You either return a value, or return `Tribe__Context::NOT_FOUND` to indicate a value was not found.
`false` does not mean "not found" in the Context, it means "false".

File: src/Tribe/Service_Providers/Context.php [hunk 2]
```diff
@@ -264,6 +273,11 @@ static function ( $post_type ) {
 				],
 				'event_post_type'      => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( TEC::POSTTYPE, false ) );
```
Comment (lucatume) [inline src/Tribe/Service_Providers/Context.php]: Return `Tribe__Context::NOT_FOUND` here, not `false` unless you really mean `false`.
If this sub-location does not return `Tribe__Context::NOT_FOUND`, then the lookup of the location will stop.

File: src/Tribe/Service_Providers/Context.php [hunk 3]
```diff
@@ -274,6 +288,11 @@ static function ( $post_type ) {
 				],
 				'venue_post_type'      => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( Venue::POSTTYPE, false ) );
```
Comment (lucatume) [inline src/Tribe/Service_Providers/Context.php]: Return `Tribe__Context::NOT_FOUND` here, not `false` unless you really mean `false`.
If this sub-location does not return `Tribe__Context::NOT_FOUND`, then the lookup of the location will stop.

File: src/Tribe/Service_Providers/Context.php [hunk 4]
```diff
@@ -284,6 +303,11 @@ static function ( $post_type ) {
 				],
 				'organizer_post_type'  => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( Organizer::POSTTYPE, false ) );
```
Comment (lucatume) [inline src/Tribe/Service_Providers/Context.php]: Return `Tribe__Context::NOT_FOUND` here, not `false` unless you really mean `false`.
If this sub-location does not return `Tribe__Context::NOT_FOUND`, then the lookup of the location will stop.

File: src/Tribe/Service_Providers/Context.php [hunk 5]
```diff
@@ -259,6 +259,15 @@ static function ( $data ) {
 				],
 				'tec_post_type'        => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return count( array_filter( [
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 73d50a3c7e17b3fc464bb5ed7b45af14</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 6]
```diff
@@ -259,6 +259,15 @@ static function ( $data ) {
...
+									! empty( tribe_get_request_var( TEC::POSTTYPE, false ) ),
+									! empty( tribe_get_request_var( Venue::POSTTYPE, false ) ),
+									! empty( tribe_get_request_var( Organizer::POSTTYPE, false ) ),
+								] ) );
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: a18c426a3c6acbe893873f97132c28b6</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 7]
```diff
@@ -259,6 +259,15 @@ static function ( $data ) {
...
+							}
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Each array item in a multi-line array declaration must end in a comma
* :no_entry_sign: Comma required after last value in array declaration

<sub>hash: 47e29295cd5ac34311c1ba358269a350</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 8]
```diff
@@ -274,6 +283,11 @@ static function ( $post_type ) {
 				],
 				'event_post_type'      => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( TEC::POSTTYPE, false ) );
+							}
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Each array item in a multi-line array declaration must end in a comma
* :no_entry_sign: Comma required after last value in array declaration

<sub>hash: 034aaf95908bdab7fb4d6eba61569f0a</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 9]
```diff
@@ -284,6 +298,11 @@ static function ( $post_type ) {
 				],
 				'venue_post_type'      => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( Venue::POSTTYPE, false ) );
+							}
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Each array item in a multi-line array declaration must end in a comma
* :no_entry_sign: Comma required after last value in array declaration

<sub>hash: 62810cdbb2773eee3368931a3ca0b158</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 10]
```diff
@@ -294,6 +313,11 @@ static function ( $post_type ) {
 				],
 				'organizer_post_type'  => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( Organizer::POSTTYPE, false ) );
+							}
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Each array item in a multi-line array declaration must end in a comma
* :no_entry_sign: Comma required after last value in array declaration

<sub>hash: f2076fbff048be996241c23ed9034a20</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 11]
```diff
@@ -259,21 +259,39 @@ static function ( $data ) {
...
+						Tribe__Context::FUNC          => [
+							static function () {
+								$found = [
+									! empty( tribe_get_request_var( TEC::POSTTYPE, false ) ),
+									! empty( tribe_get_request_var( Venue::POSTTYPE, false ) ),
+									! empty( tribe_get_request_var( Organizer::POSTTYPE, false ) ),
+								];
+								$found = array_filter( $found );
+
+								return count( $found ) ?: Tribe__Context::NOT_FOUND;
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: :no_entry_sign: Using short ternaries is not allowed

<sub>hash: b4de26a2a672e063cea2fc6ef2cb691d</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 12]
```diff
@@ -259,21 +259,39 @@ static function ( $data ) {
...
+						],
 						Tribe__Context::LOCATION_FUNC => [
 							'post_type',
 							static function ( $post_type ) {
-								return count(
+								$found = count(
 									array_intersect(
 										(array) $post_type,
-										[ TEC::POSTTYPE, Venue::POSTTYPE, Organizer::POSTTYPE ]
+										[ TEC::POSTTYPE, Venue::POSTTYPE, Organizer::POSTTYPE, ]
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: * :no_entry_sign: Comma not allowed after last value in single-line array declaration
* :no_entry_sign: Comma not allowed after last value in single-line array declaration

<sub>hash: 65aaf622924c2792cdf90d01c81b5021</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 13]
```diff
@@ -259,21 +259,39 @@ static function ( $data ) {
...
 									)
 								);
+								return $found ?: Tribe__Context::NOT_FOUND;
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: :no_entry_sign: Using short ternaries is not allowed

<sub>hash: eee6c61c81fa1c84775a4e8d8dca4b97</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 14]
```diff
@@ -259,21 +259,39 @@ static function ( $data ) {
...
 							},
 						],
 					],
 				],
 				'event_post_type'      => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( TEC::POSTTYPE, false ) ) ?: Tribe__Context::NOT_FOUND;
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: :no_entry_sign: Using short ternaries is not allowed

<sub>hash: 3292b9e739d10f509436adcb82ff0897</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 15]
```diff
@@ -284,6 +302,11 @@ static function ( $post_type ) {
 				],
 				'venue_post_type'      => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( Venue::POSTTYPE, false ) ) ?: Tribe__Context::NOT_FOUND;
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: :no_entry_sign: Using short ternaries is not allowed

<sub>hash: 252f2a661e7328091d253fc702c8911c</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 16]
```diff
@@ -294,6 +317,11 @@ static function ( $post_type ) {
 				],
 				'organizer_post_type'  => [
 					'read' => [
+						Tribe__Context::FUNC          => [
+							static function () {
+								return ! empty( tribe_get_request_var( Organizer::POSTTYPE, false ) ) ?: Tribe__Context::NOT_FOUND;
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php]: :no_entry_sign: Using short ternaries is not allowed

<sub>hash: f677f654ca87bf50b9933d325bacdd9f</sub>

**Pull request merged**